### PR TITLE
DEP-334: configure jetty options

### PR DESCRIPTION
**Pull request by xiwenc**
Manually tested:

```
cf set-env 675bc0be-de1a-4cc9-b067-78f0c8540a82 JETTY_CONFIG '{"runtime_max_threads": 100}'

```

and restarted the app:

screenshot 2018-06-28 at 13 33 24

### CODE REVIEW

File: start.py [hunk 1]
```diff
@@ -293,6 +293,19 @@ def set_jvm_memory(m2ee_section, vcap, java_version):
...
 
+def set_jetty_config(m2ee):
+    jetty_config_json = os.environ.get('JETTY_CONFIG')
+    if not jetty_config_json:
+        return None
+    try:
+        jetty_config = json.loads(jetty_config_json)
+        jetty = m2ee.config._conf['m2ee']['jetty']
+        jetty.update(jetty_config)
+        logger.debug('Jetty configured: %s ' % json.dumps(jetty))
```
Comment (joshp123) [inline start.py]: Use the comma-based syntax for logging because otherwise it creates the string even when not required:
https://stackoverflow.com/a/34634301/2230226

tl;dr:
`logger.debug('foo is %s', foo)` vs `logger.debug('foo is %s' % foo)`
In #1 the string is lazily constructed, #2 is not.

File: tests/usecase/test_jetty_config.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import basetest
+import json
+
+
+class TestCaseJettyConfig(basetest.BaseTest):
```
Comment (joshp123) [inline tests/usecase/test_jetty_config.py]: Can we also add tests for the unhappy paths?
- Invalid JSON in the `JETTY_CONFIG` environment variable
- (maybe) `JETTY_CONFIG` set but empty

File: start.py [hunk 3]
```diff
@@ -293,6 +293,19 @@ def set_jvm_memory(m2ee_section, vcap, java_version):
...
+    except Exception as e:
+        logger.warning('Failed to configure jetty: ' + str(e))
```
Comment (joshp123) [inline start.py]: You can do this better in the following way:
```
logger.warning("Failed to configure jetty.", exc_info=True)
```
Because that includes not only the string representation of the exception, but also the traceback.

File: tests/usecase/test_jetty_config.py (hunk 2 above)
Comment (xiwenc) [reply tests/usecase/test_jetty_config.py]: added extra case with invalid json

**Comment by pommi** [comment]
This was DEP-344 btw. not DEP-334
